src/extension/commandPaletteHandler.ts

```diff
@@ -195,269 +170,206 @@ export class CommandPaletteHandler {
         TipNotificationService.getInstance().setKnownDateForFeatureById("debuggingRNWAndMacOSApps");
         const additionalPackagesToCheck: ParsedPackage[] = [
             REACT_NATIVE_PACKAGES.REACT_NATIVE_MACOS,
         ];
-        return this.selectProject().then((appLauncher: AppLauncher) => {
-            TargetPlatformHelper.checkTargetPlatformSupport(PlatformType.macOS);
-            return ProjectVersionHelper.getReactNativePackageVersionsFromNodeModules(
-                appLauncher.getOrUpdateNodeModulesRoot(),
-                additionalPackagesToCheck,
-            ).then(versions => {
-                appLauncher.setReactNativeVersions(versions);
-                return this.executeCommandInContext(
-                    "runMacOS",
-                    appLauncher.getWorkspaceFolder(),
-                    () => {
-                        const platform = <MacOSPlatform>(
-                            this.createPlatform(appLauncher, PlatformType.macOS, MacOSPlatform)
-                        );
-                        return (
-                            platform
-                                .beforeStartPackager()
-                                .then(() => {
-                                    return platform.startPackager();
-                                })
-                                .then(() => {
-                                    // Set the Debugging setting to disabled, because in macOS it persists across runs of the app
-                                    return platform.disableJSDebuggingMode();
-                                })
-                                // eslint-disable-next-line @typescript-eslint/no-empty-function
-                                .catch(() => {}) // If setting the debugging mode fails, we ignore the error and we run the run ios command anyways
-                                .then(() => {
-                                    return platform.runApp();
-                                })
-                        );
-                    },
+        const appLauncher = await this.selectProject();
+        TargetPlatformHelper.checkTargetPlatformSupport(PlatformType.macOS);
+        const versions = await ProjectVersionHelper.getReactNativePackageVersionsFromNodeModules(
+            appLauncher.getOrUpdateNodeModulesRoot(),
+            additionalPackagesToCheck,
+        );
+        appLauncher.setReactNativeVersions(versions);
+        return await this.executeCommandInContext(
+            "runMacOS",
+            appLauncher.getWorkspaceFolder(),
+            async () => {
+                const platform = <MacOSPlatform>(
+                    this.createPlatform(appLauncher, PlatformType.macOS, MacOSPlatform)
                 );
-            });
-        });
+                try {
+                    await platform.beforeStartPackager();
+                    await platform.startPackager();
+                    await platform.disableJSDebuggingMode();
```

Comment:
```suggestion
                    // Set the Debugging setting to disabled, because in macOS it persists across runs of the app
                    await platform.disableJSDebuggingMode();
```